### PR TITLE
<Stepper/> - generate new  component with props validation

### PR DESCRIPTION
**Pull request by mosheKerbel**

### CODE REVIEW

File: src/beta/Stepper/test/Stepper.spec.js [hunk 1]
```diff
@@ -1,22 +1,58 @@
...
+      render(<Stepper active={1} steps={[{}]} />);
+      expect(consoleErrorSpy).toHaveBeenCalledWith(
+        expect.stringContaining(
+          'Warning: Failed prop type: The prop `steps[0].text` is marked as required in `Stepper`, but its value is `undefined`',
+        ),
+      );
+    });
+
+    it('should warn when steps length is less than 2', async () => {
+      expect(true).toEqual(true); // todo: impl
```
Comment (mosheKerbel) [inline src/beta/Stepper/test/Stepper.spec.js]: i am considering validating the steps size.
1 < stepSize < 8
can do it as a part of the props-validation by using a HOC as in this example
https://www.ian-thomas.net/custom-proptype-validation-with-react/

no other component is using it. will need to add it as a HOC and add tests.
if i do it i prefer adding it as a task and get back to it later.

WDYT?
@shlomitc

**Comment by mosheKerbel** [comment]
> @mosheKerbel are you sure this shouldn't be a compound component? i imagine this shape will grow in future, require tooltips and stuff

It shouldn't. You can take a look here, the component is very restricted.
https://app.zeplin.io/project/5c8f6f1f9f489505cac756dc/screen/5cc8015405f4f16514851df4

**Comment by argshook** [comment]
> > @mosheKerbel are you sure this shouldn't be a compound component? i imagine this shape will grow in future, require tooltips and stuff
> 
> It shouldn't. You can take a look here, the component is very restricted.
> [app.zeplin.io/project/5c8f6f1f9f489505cac756dc/screen/5cc8015405f4f16514851df4](https://app.zeplin.io/project/5c8f6f1f9f489505cac756dc/screen/5cc8015405f4f16514851df4)

if only i get a nickel for each "shouldn't" that later turns into "should" :)

i would implement with compound idea, but ignore me

**Comment by mosheKerbel** [comment]
> if only i get a nickel for each "shouldn't" that later turns into "should" :)
> 
> i would implement with compound idea, but ignore me

i kind of disagree because it contradicts with the idea of this component spec.

**Comment by mosheKerbel** [comment]
@nitayneeman @shlomitc

File: src/Stepper/Stepper.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    this._validateNumberOfSteps(props.steps);
+  }
+
+  componentDidUpdate() {
+    this._validateNumberOfSteps(this.props.steps);
+  }
+
+  _validateNumberOfSteps(steps) {
+    if (!steps || !steps.length || !steps.length < 2 || steps.length > 7) {
```
Comment (nitayneeman) [inline src/Stepper/Stepper.js]: I would export these magic numbers to variables and reuse them

File: src/Stepper/Stepper.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      console.error(
+        'Warning: Failed prop: The prop `steps` in `Stepper` has to be an array in the size of 2 to 7.',
+      );
+    }
+  }
+
+  _renderStep = (step, idx) => {
```
Comment (nitayneeman) [inline src/Stepper/Stepper.js]: As a convention - I usually place the private methods at the bottom (after `render`)

File: src/Stepper/docs/index.story.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+          divider(),
+
+          title('Examples'),
+
+          code({
+            title: 'Stepper',
+            compact: true,
+            source:
+              '<div style={{padding: "20px 20px", backgroundColor: "white"}}>' +
```
Comment (nitayneeman) [inline src/Stepper/docs/index.story.js]: Why not template strings?

File: src/Stepper/test/Stepper.spec.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    consoleErrorSpy = jest
+      .spyOn(global.console, 'error')
+      .mockImplementation(jest.fn());
+  });
+
+  afterEach(() => {
+    consoleErrorSpy.mockRestore();
+  });
+
+  describe('Stepper', () => {
```
Comment (nitayneeman) [inline src/Stepper/test/Stepper.spec.js]: Maybe it's worth to add a test which verifies that when you pass X number of steps it actually renders the correct number of steps

File: src/Stepper/Stepper.js (hunk 3 above)
Comment (mosheKerbel) [reply src/Stepper/Stepper.js]: actually i'm used to another convention where the main render is in the bottom of the file and the other private renders above it :)
do we have any guideline for wsr?

File: src/Stepper/test/Stepper.spec.js (hunk 5 above)
Comment (mosheKerbel) [reply src/Stepper/test/Stepper.spec.js]: all this kind of tests will be in the pending next PR

**Pull request closed without merging**